src/Concordium/Client/Types/Contract/Parameter.hs

```diff
@@ -244,6 +256,23 @@ putJSONUsingSchema typ json = case (typ, json) of
         pure $ putLen <> putJSONFields'
     _ -> Left [i|#{obj} had too many fields. It should contain a single variant of the following enum:\n#{showPrettyJSON enum}.|]
 
+  (tEnum@(TaggedEnum variants), AE.Object obj) -> case KM.toList obj of
```

Comment:
I want to have a look at this again when it has been aligned with cargo-concordium 😊